lib/roger/job.ex

```diff
@@ -133,11 +119,24 @@ defmodule Roger.Job do
   def create(module, args \\ [], id \\ generate_job_id()) when is_atom(module) do
     keys =
       ~w(queue_key execution_key)a
-      |> Enum.map(fn(prop) -> {prop, Kernel.apply(module, prop, [args])} end)
+      |> Enum.map(fn(prop) ->
+        if function_exported?(module, prop, 1) do
```

Comment:
Is there no way around all the `function_exported?/3` checks?

Reply:
It's a pity they didn't implement a cleaner way for that I looked up how elixir developers handled it in genserver optional callbacks.

```
 Usually, behaviour modules know if they should call those callbacks based on configuration, or they check if the callbacks are defined with `function_exported?/3` or `macro_exported?/3`.
```

Reply:
And using something like `Code.ensure_loaded` when creating a job?

Reply:
Yeah that might be good check to add. So apps that don't fully embed everything always work.